lib/OpenQA/WebAPI/Plugin/REST.pm

```diff
@@ -2,49 +2,33 @@
 # SPDX-License-Identifier: GPL-2.0-or-later
 
 package OpenQA::WebAPI::Plugin::REST;
-use Mojo::Base 'Mojolicious::Plugin';
+use Mojo::Base 'Mojolicious::Plugin', -signatures;
 
-use Scalar::Util ();
-use Carp ();
-
-sub register {
-
-    my ($self, $app, $config) = @_;
+use Scalar::Util 'blessed';
+use Carp 'croak';
 
+sub register ($self, $app, $config) {
     # special anchor tag with data-method
     $app->helper(
-        action_link => sub {
-            my ($self, $method, $content) = (shift, shift, shift);
+        action_link => sub ($c, $method, $content, @args) {
             my $url = $content;
-
-            if ($self->is_operator) {
-                # Content
-                unless (ref $_[-1] eq 'CODE') {
-                    $url = shift;
-                    push @_, $content;
-                }
-
-                Carp::croak "url is not a url"
-                  unless Scalar::Util::blessed $url && $url->isa('Mojo::URL');
-
-                return $self->tag('a', href => $url, 'data-method' => $method, @_);
-            }
-            else {
-                return '';
+            return '' unless $c->is_operator;
+            # Content
+            unless (ref $args[-1] eq 'CODE') {
+                $url = shift @args;
+                push @args, $content;
```

Comment:
this code seems to be uncovered. Anyone has an idea what it could be useful for? Or should we just remove it?

Reply:
Not sure. So Git blame doesn't reveal anything?

Reply:
Well, @aaannz has introduced it, we never changed it and it's not called in tests. What behavior could this be for?

Reply:
You mean @ancorgs ( commit https://github.com/os-autoinst/openQA/commit/34d51fb0c3c20d6a6e76bf13f5ae7e6ae52bd919 ). I have no idea what it is of and if it is still valid.

Reply:
Well, I meant the refactoring in 04c265d57 but looks like you found a predecessor. I guess I will just remove that code.